### PR TITLE
T ronmoneta/java sample app

### PR DESCRIPTION
**Pull request by ronmonetaMicro**
#### Pull Request Description

Re structured sample app

---

#### Future Release Comment
_[Add description of your change, to include in the next release]_
_[Delete any or all irrelevant sections, e.g. if your change does not warrant a release comment at all]_

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- None

### CODE REVIEW

File: quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.UUID;
+
+/**
+ * SourceType - represents the type of files used for ingestion
+ */
+enum SourceType {
+    localFileSource, blobSource
```
Comment (AsafMah) [inline quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java:23]: java enum members are usually in SHOUTY_CASE

File: quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+enum AuthenticationModeOptions {
+    userPrompt("UserPrompt"), managedIdentity("ManagedIdentity"), appKey("AppKey"), appCertificate("AppCertificate");
+
+    private final String mode;
+
+    AuthenticationModeOptions(String mode) {
+        this.mode = mode;
+    }
+
+    public String getMode() {
```
Comment (AsafMah) [inline quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java:38]: You don't need a getter for something that's never called

File: quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+
+    /**
+     * Second phase - The ingestion process
+     *
+     * @param config       ConfigJson object containing the SampleApp configuration
+     * @param kustoClient  Client to run commands
+     * @param ingestClient Client to ingest data
+     */
+    private static void ingestion(ConfigJson config, Client kustoClient, IngestClient ingestClient) {
+        for (ConfigData data_source : config.getData()) {
```
Comment (AsafMah) [inline quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java:491]: camelCase

**Review by AsafMah — COMMENTED** [review]
Looks very good overall, fix these small things, I'll look over again and approve.

File: quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java (hunk 1 above)
Comment (ronmonetaMicro) [reply quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java:23]: Wrote it like that to automatically identify it from JSON. should i look for ways to convert like we did in python?

Comment (AsafMah) [reply quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java:23]: Does giving them string values don't work? (Like you did in the next enum)

**Review by AsafMah — APPROVED** [review]

**Pull request merged**